nav2_system_tests/src/system/test_system_launch.py

```diff
@@ -40,15 +43,27 @@ def generate_launch_description():
     bringup_dir = get_package_share_directory('nav2_bringup')
     params_file = os.path.join(bringup_dir, 'params', 'nav2_params.yaml')
 
-    # Replace the `use_astar` setting on the params file
-    param_substitutions = {
-        'planner_server.ros__parameters.GridBased.use_astar': os.getenv('ASTAR')}
+    # Replace the default parameter values for testing special features
+    # without having multiple params_files inside the nav2 stack
+    context = LaunchContext()
+    param_substitutions = {}
+
+    if (os.getenv('ASTAR') == "True"):
+        param_substitutions.update({'use_astar': "True"})
+        print("ASTAR set True")
+
+    if (os.getenv('GROOT_MONITORING') == "True"):
+        param_substitutions.update({'enable_groot_monitoring': "True"})
+        print("GROOT_MONITORING set True")
+
+    print
     configured_params = RewrittenYaml(
         source_file=params_file,
         root_key='',
         param_rewrites=param_substitutions,
         convert_types=True)
 
+    new_yaml = configured_params.perform(context)
```

Comment:
What is this?

Reply:
Without the perform function of the RewrittenYaml Class-object, the perform is not triggered. This creates a new temporary yaml file under /tmp . So new_yaml should only be a string to a temporal yaml file.

The new_yaml variable then gets used to start the test. As far as I could tell I'm currently the only one utilizing this in navigation2. There are some other usage of RewrittenYaml, but I'm pretty confident that the desired substitution does not get performed there. That were my testing results while trying to get RewrittenYaml to properly work and not rewriting it.

Reply:
Fine for you?